### ISSUE TITLE
Footnote should follow text color

### ISSUE THREAD

**Issue by bph** [WordPress/gutenberg#51895]
### Description

I added a footnote to a text that as displayed in white with a blue background. When I added a footnote I didn't see the [1] as it didn't follow the text color of the paragraph and was using the same blue as the background color. With further testing it revealed that it would be a bad color contrast with any text that doesn't have a very light background. 

### Step-by-step reproduction instructions

Create a paragraph with text color white and background color blue
Write a few characters
Use the inline download menu to add a footnote. 
Add the note

### Screenshots, screen recording, code snippet

<img width="174" alt="Screenshot 2023-06-26 at 11 09 35" src="https://github.com/WordPress/gutenberg/assets/39980/d72b8944-4d7f-4a67-a865-9a9ceed19374">
<img width="230" alt="Screenshot 2023-06-26 at 11 09 12" src="https://github.com/WordPress/gutenberg/assets/39980/352a12c1-1a54-4060-9938-c5d3ce4a9893">

### Environment info

GB-nightly: 16.2.20230626
WordPress 6.3-alpha-55991 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
I just tested as well and was able to confirm in TT3, but I think this is due to the fact that the theme does not define a Link color. If you set a link color in Global Style, or at the block level, the footnote links respect that color. 

![Image](https://github.com/WordPress/gutenberg/assets/4832319/369dda50-124f-4c1b-939b-60b10bee89b7)

**Comment by bph** [comment]
@ndiego Thanks for your guidance. I can confirm. When a link color is defined the footnote link uses it to. 
We can close this issue now. As it works as designed. I do have the question, why TT3 doesn't have a link color defined. But that's outside the scope here. :-)

**Comment by ndiego** [comment]
> I do have the question, why TT3 doesn't have a link color defined. But that's outside the scope here. :-)

Yeah that is a great question, perhaps to make it as simple as possible. Links are underlined in the theme, but a color change would have been helpful as well.

**Comment by bph** [comment]
Is it possible within a theme to have the link color follow the text color, if there isn't a separate link color defined?

**Comment by ndiego** [comment]
> I just tested as well and was able to confirm in TT3, but I think this is due to the fact that the theme does not define a Link color.

Just wanted to provide a correction here for those that find this issue. TT3 **does** define a link color, I had inadvertently been using a child theme in my testing that was heavily modified. 🤦‍♂️
